src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas2/ApplicationsController.kt

```diff
@@ -136,6 +136,26 @@ class ApplicationsController(
     return ResponseEntity.ok(getPersonDetailAndTransform(updatedApplication))
   }
 
+  @Transactional
```

Comment:
On this commits message:

> it's not a hard
DELETE request because the Entity will remain in
our db, but its also not really an UPDATE request
because the entity is being deleted from the UI.

Agree on it not being a DELETE but I do think it is an UPDATE request at least as far as the API is concerned. If I recall we hit problems when trying to reuse the existing update endpoint, it was set up as a PUT instead of a PATCH despite it modifying only the `data` field. That meant we couldn't easily extend it to handle our new type of update. 

Instead of initiating _another_ migration for that endpoint given our remaining time, we decided to bring up a new action and leave the old for another day. If we'd had more time we could have converted that endpoint into a PATCH to keep it more conventional, changed the frontends request and then extended the action with our new behaviour.

This context may be worth throwing into the commit if I've remembered it right.

Reply:
Mind you, having now read the contents of the action below it has enough logic to likely merit extracting into it's own action anyway, for simplicity, readability etc.